### ISSUE TITLE
Implement end-to-end transducer tracking

### ISSUE THREAD

**Issue by ebrahimebrahim** [OpenwaterHealth/SlicerOpenLIFU#149]
## Goal

Make the "Run transducer tracking" button run transducer tracking.

There are actually several steps involved in transducer tracking and we will later want to break down the workflow into a refined step-by-step process, but for now in this issue we just want to get the blind end-to-end workflow working smoothly.

We will also remove the temporary controls that were created for the initial "mimimal" transducer tracking workflow.

## Details

After running transducer tracking, the algorithm gives us two transforms as output: one mapping the photoscan into MRI space, and one mapping the transducer registration surface into photoscan space.

Whether in treatment/session mode or in manual mode:
- these transforms should become slicer transform nodes
- the photoscan textured model is loaded into the slicer scene
- the transforms should get applied so that
  - the photoscan is transformed to the MRI space
  - the transducer is also transformed to the MRI space. So the current transducer transform node gets updated with the appropriate composition of the two matrices output by the algorithm, but we also save a separate transform node to serve as a snapshot of this matrix (Similar principle to what we do in #145 for virtual fitting).

In treatment/session mode:
- the result of the transducer tracking is saved into the session as well: the session stores a list of “transducer tracking results”, each of which is a photoscan ID and two transforms, and we just use the underlying openlifu functionality introduced in https://github.com/OpenwaterHealth/OpenLIFU-python/issues/182 to append a transducer tracking result

### Skin mesh from mri

Part of the transducer tracking is to take the raw volume input and generate a skin mesh. We will use openlifu to do this once  https://github.com/OpenwaterHealth/OpenLIFU-python/issues/149 is done. This will replace the temporary measure #152 

### Initialization of transducer transform

Update: We just found out that the transducer tracking algorithm needs a transducer transform to use as initialization to help it do the registration between transducer-registration-surface and photoscan.

In manual (no session) mode we can just use the transducer transform, so it is up to the user to move around the transducer to a good initialization. 
In treatment (session) mode we can use as candidates all the virtual fit candidates that are in the session, from https://github.com/OpenwaterHealth/OpenLIFU-python/issues/179.

## Things yet to clarify

A few things are unclear to me, and maybe we can think about them as we get into this item.

How should the new transform nodes be handled? Are they "openlifu objects" to be listed in the list of openlifu objects? Or are they affiliated objects of other things (like how the "current transducer transform node" is affiliated to the SlicerOpenLIFUTransducer)?

What happens when you load a session that already has transducer tracking results in it? Does it show up in the session status view that there is already a transducer tracking result?

Is there a way to load a previous transducer tracking result? Should that be a separate issue?

## Dependencies

Depends on #143 
~~Depends on #138~~ See [below](https://github.com/OpenwaterHealth/SlicerOpenLIFU/issues/149#issuecomment-2622575852).
Depends on #160
Depends on #148 
Depends on #157 
Depends on https://github.com/OpenwaterHealth/OpenLIFU-python/issues/182
Depends on https://github.com/OpenwaterHealth/OpenLIFU-python/issues/179

**Comment by ebrahimebrahim** [comment]
New way of approaching this: Implement the manual transducer tracking protocol _first_.

Notes here: https://docs.google.com/document/d/1E9hJlHqPt-KLCHQNYtSKhpBEe1FLfrtNmtYeeNP1Azo/edit?tab=t.0#heading=h.k6n9lcitkf0o

We will gradually automate the 3 major time consuming manual steps:
- Putting facial landmarks on the photoscan
- Putting landmarks on the mri skin seg
- Putting transducer landmarks on the photoscan

**Comment by ebrahimebrahim** [comment]
Remember to also remove the temporary "select skin segmentation model" dropdown and replace it by the skin segmentation introduced in https://github.com/OpenwaterHealth/OpenLIFU-python/pull/190

**Comment by ebrahimebrahim** [comment]
@sadhana-r  Peter has provided a list of which aspects of "manual transducer tracking" they hope to have by April 1:

> 1. Extraction of the skin surface from the MRI
> 2. Guided marking of the Eyes and Nasion
> 3. Guided marking of the Eyes and Nasion on the Photoscan
> 4. Fiducial and Surface Registration (similarity) of the Photoscan to the MRI
> 5. Surface registration of the transducer to the Photoscan, using the virtual fit location to initiate the search.

And this is the doc where we have the latest written summary of the manual workflow: https://docs.google.com/document/d/1rWI5ZUjMTd-zCz1qryXEaxuapIDLFfWw9b7KF4eu-tA/edit?usp=drive_link

**Comment by ebrahimebrahim** [comment]
To add some comments from today's meeting, for our first end-to-end transducer tracking:
- Fiducial based registration only is sufficient for the photoscan <--> MRI registration. The ICP step can be a placeholder that returns the identity transform for example
- For the transducer <--> photoscan registration, it is okay for now to have a placeholder in which the user is just given thhe transform handles to manually move the transducer into place.

**Comment by sadhana-r** [comment]
This issue has been broken down into the following, smaller issues:

- #199 
- #200 
- #201
- #202
- #203
- #196
